feat(world): enable config namespaces key (#2880)
latticexyz/mud: .changeset/violet-flowers-poke.md

---

> "@latticexyz/store": patch

Reviewer: ```suggestion
```
there's no real user-facing store changes, just conditionally defaulting `indexFilename` to `false` if `namespaces` is used, so would probably opt to exclude this changeset from this description

I could see leaving this here if store had `namespaces` support too, but we haven't migrated that yet

"@latticexyz/world": patch
---

> Enabled the `namespaces` key in the config. This enables declaring resources in different namespaces in a single MUD config.
> 
> For example:

Reviewer: ```suggestion
You can now configure resources across multiple namespaces in a single MUD config using the `namespaces` key.

Note that once you start using `namespaces` config option, you will need to move your existing tables, systems, etc. under the corresponding namespace within `namespaces`.
```

```ts
export default defineWorld({
  namespaces: {
    game: {
      tables: {
        Health: {
          schema: {
            player: "address",
            value: "uint32",
          },
          key: ["player"],
        },
        Position: {
          schema: {
            player: "address",
            x: "int32",
            y: "int32",
          },
          key: ["player"],
        },
      },
    },

>     somePlugin: {

Reviewer: good to give folks a concrete example rather than "some" abstract one
```suggestion
    scoreboardPlugin: {
```

Author: that's the name i was looking for 🙇

Author: This namespace name is actually too long, it causes the project to fail in in unexpected ways

Reviewer: we should fix that and/or add it to validation

Author: Working on it in https://github.com/latticexyz/mud/pull/2897
